tests/Query/Filter/ReferenceDocs/FilterReference/DateFieldReference.test.ts

```diff
@@ -11,7 +11,8 @@ window.moment = moment;
 describe('explain', () => {
     beforeEach(() => {
         jest.useFakeTimers();
-        jest.setSystemTime(new Date('2023-04-19'));
+        // Use moment to get a timezone aware Date object
+        jest.setSystemTime(moment('2023-04-19', 'YYYY-MM-DD').toDate());
```

Comment:
> The approach in #1851 is probably the better call. I can reimplement this PR with that fix, or just let it get merged in with #1851

If I understand it correctly, you wrote this code because the current behaviour is inconveniencing people in some time zones...

I am unsure how long it will be until that other PR will be merged, so your suggestion of reimplementing this PR with that fix is good.

From memory, I suspect that you cannot cherry-pick the commit in that other PR as I think it was part of a larger change...

If that's true, recreating that change in this PR by hand is fine...

Could you please update the PR description and summary too...

/CC @sytone for info...

Reply:
> Could you please update the PR description and summary too...

(here, I mean...)